### ISSUE TITLE
APM mappings changes should be shown in release notes, improve guidance on customizations on the documentation

### ISSUE THREAD

**Issue by lucabelluccini** [elastic/apm-server#14813]
This issue is to track 2 possible coupled items:
- Whenever we change any field type in the built-in APM assets, it should show up in the release notes as breaking change
- In the `@custom` doc section of our O11y docs, we should recommend to not override the fields we declare but prefer moving the fields to a different "namespace" (e.g. `custom.myfield` as it's less likely this will trigger a mapping conflict). In order to do so, the user would need to introduce an ingest pipeline which extracts the relevant fields and moves / processes them and finally stores them into `custom.<fieldname>`, plus add a mapping to the <apm data stream>@custom` component template (example below)

In short, if a customer wants to extract a specific HTTP header, instead of overriding `http.request.myheader` mappings, it would be better to:
- define a `traces-apm@custom` ingest pipeline where we take `http.request.myheader` and store it on `custom.myheader` + define a `traces-apm@custom` component template to specify the mapping of `custom.myheader` to whatever type the user wants
